vault/router.go

```diff
@@ -10,11 +10,17 @@ import (
 
 	"github.com/armon/go-metrics"
 	"github.com/armon/go-radix"
-	"github.com/hashicorp/vault/helper/consts"
 	"github.com/hashicorp/vault/helper/salt"
 	"github.com/hashicorp/vault/logical"
 )
 
+var (
+	whitelistedHeaders = []string{
+		"X-Vault-Kv-Client",
+		"x-vault-kv-client",
```

Comment:
I played around a bit with HTTP/2 request headers to see how those are set into http.Request.Header by the stdlib. It looks like we don't have to check for the lowercased value because the value gets converted into a MIME header whenever it's serialized into Go's http.Request,. 

Since `logical.Request.Header` is simply `http.Request.Header` (set in `buildLogicalRequest`), the same should apply. It looks that this is the same for HTTP/2 requests as well. It might be somewhat related to [this thread](https://github.com/golang/go/issues/5022).

The block below is a sample request to a http2 server hosted by Go, and that particular endpoint returns the request info. Passing in a lowercased header yields a MIME request header as far as `http.Request.Header` goes:
```
curl -H "foo: bar" -H "foo: baz" --http2 https://http2.golang.org/reqinfo
Method: GET
Protocol: HTTP/2.0
Host: http2.golang.org
RemoteAddr: <redacted>:50730
RequestURI: "/reqinfo"
URL: &url.URL{Scheme:"", Opaque:"", User:(*url.Userinfo)(nil), Host:"", Path:"/reqinfo", RawPath:"", ForceQuery:false, RawQuery:"", Fragment:""}
Body.ContentLength: 0 (-1 means unknown)
Close: false (relevant for HTTP/1 only)
TLS: &tls.ConnectionState{Version:0x303, HandshakeComplete:true, DidResume:false, CipherSuite:0xc02b, NegotiatedProtocol:"h2", NegotiatedProtocolIsMutual:true, ServerName:"http2.golang.org", PeerCertificates:[]*x509.Certificate(nil), VerifiedChains:[][]*x509.Certificate(nil), SignedCertificateTimestamps:[][]uint8(nil), OCSPResponse:[]uint8(nil), TLSUnique:[]uint8{0x32, 0x21, 0xf7, 0x51, 0xaf, 0xaa, 0x29, 0x7c, 0x25, 0x16, 0xe1, 0xe8}}

Headers:
Accept: */*
Foo: bar
Foo: baz
User-Agent: curl/7.54.0
```